### PR TITLE
[FEATURE: UIPT-25] - Carousel

### PR DESCRIPTION
**Pull request by IrynaMukhina**

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -1,11 +1,57 @@
...
+        PopularListComponent,
+        ProductCarouselComponent,
+      ],
+      imports: [
+        BrowserModule,
+        BrowserAnimationsModule,
+        HttpClientModule,
+        FontAwesomeModule
+      ],
+      providers: [HttpService,
```
Comment (AlexLarusso) [inline src/app/app.component.spec.ts]: It is useless spec. Please remove

File: src/app/product-carousel/product-carousel.component.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<div class="carousel__wrapper">
```
Comment (AlexLarusso) [inline src/app/product-carousel/product-carousel.component.html]: I think carousel should be in shared folder

File: src/app/product-carousel/product-carousel.component.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  }
+
+  public toggleButtonsState(): void {
+    const maxPosition = -this.itemWidth * (this.productArray.length - this.visibleNum);
+
+    this.canMoveToPrev = this.position !== 0;
+    this.canMoveToNext = this.position !== maxPosition;
+  }
+
+  public moveTo(direction: string): void {
```
Comment (AlexLarusso) [inline src/app/product-carousel/product-carousel.component.ts]: Use 2 methods instead if it is possible:
moveToNext
moveToPrev

File: src/app/product-carousel/product-carousel.component.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      this.position = newPosition >= possiblePosition ? possiblePosition : newPosition;
+    }
+
+    this.productContainer.style.marginLeft = this.position + this.additionalScroll + 'px';
+
+    this.toggleButtonsState();
+  }
+
+  public ngAfterViewChecked(): void {
+    const [_, ...rest] = this.products.nativeElement.childNodes;
```
Comment (AlexLarusso) [inline src/app/product-carousel/product-carousel.component.ts]: Why do you need this part? `[_,`

Comment (IrynaMukhina) [reply src/app/product-carousel/product-carousel.component.ts]: The first element of product.childNodes is a comment, so to resolve this issue I use destructuring.
image

**Comment by IrynaMukhina** [comment]
@AlexLarusso thank you for reviewing

**Pull request merged**